packages/smooth_app/lib/pages/product/common/product_list_page.dart

```diff
@@ -106,6 +108,7 @@ class _ProductListPageState extends State<ProductListPage>
     return SmoothScaffold(
       floatingActionButton: products.isEmpty
           ? FloatingActionButton.extended(
+              heroTag: 'compare_fab_${Random(100)}',
```

Comment:
What is the reason of random hero tag, then there won't be anything to hero to or am I missing something